litestar/logging/standard.py

```diff
@@ -1,34 +1,45 @@
 from __future__ import annotations
 
 import atexit
-import sys
-from logging import StreamHandler
+from logging import Handler, LogRecord, StreamHandler
 from logging.handlers import QueueHandler, QueueListener
 from queue import Queue
 from typing import Any
 
 from litestar.logging._utils import resolve_handlers
 
-__all__ = ("QueueListenerHandler",)
+__all__ = ("LoggingQueueListener", "QueueListenerHandler")
 
 
-if sys.version_info < (3, 12):
+class LoggingQueueListener(QueueListener):
+    """Custom ``QueueListener`` which starts and stops the listening process."""
 
-    class QueueListenerHandler(QueueHandler):
-        """Configure queue listener and handler to support non-blocking logging configuration."""
+    def __init__(self, queue: Queue[LogRecord], *handlers: Handler, respect_handler_level: bool = False) -> None:
+        """Initialize ``LoggingQueueListener``.
 
-        def __init__(self, handlers: list[Any] | None = None) -> None:
-            """Initialize `?QueueListenerHandler`.
+        Args:
+            queue: The queue to send messages to
+            *handlers: A list of handlers which will handle entries placed on the queue
+            respect_handler_level: If ``respect_handler_level`` is ``True``, a handler's level is respected (compared with the level for the message) when deciding whether to pass messages to that handler
+        """
+        super().__init__(queue, *handlers, respect_handler_level=respect_handler_level)
+        self.start()
+        atexit.register(self.stop)
 
-            Args:
-                handlers: Optional 'ConvertingList'
-            """
-            super().__init__(Queue(-1))
-            handlers = resolve_handlers(handlers) if handlers else [StreamHandler()]
-            self.listener = QueueListener(self.queue, *handlers)
-            self.listener.start()
 
-            atexit.register(self.listener.stop)
+class QueueListenerHandler(QueueHandler):
+    """Configure queue listener and handler to support non-blocking logging configuration.
 
-else:
-    QueueListenerHandler = QueueHandler
+    This handler won't work with Python >= 3.12 and ``logging.config.dictConfig``. It might be deprecated in the future,
+    please use ``logging.QueueHandler`` instead.
+    """
+
+    def __init__(self, handlers: list[Any] | None = None) -> None:
+        """Initialize `QueueListenerHandler`.
+
+        Args:
+            handlers: Optional 'ConvertingList'
+        """
+        super().__init__(Queue(-1))
+        handlers = resolve_handlers(handlers) if handlers else [StreamHandler()]
+        self.listener = LoggingQueueListener(self.queue, *handlers)  # type: ignore[arg-type]
```

Comment:
mypy was returning:
```
litestar/logging/standard.py:45: error: Argument 1 to "LoggingQueueListener" has incompatible type "_QueueLike[Any]"; expected "Queue[LogRecord]"  [arg-type]
```

I don't know if I could do something better here. It seems I cannot import `_QueueLike`.

Reply:
I think this is fine. It seems that `QueueHandler.queue` isn't typed properly as `Queue[LogRecord]` internally, so nothing we can fix here.

Reply:
@provinzkraut I guess I could type `LoggingQueueListener.queue` as `Any` and remove `type: ignore[arg-type]` here. But not sure it would be better.

Reply:
I think that'd be more confusing actually